platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -969,6 +977,17 @@ int _odp_ipsec_sa_lifetime_update(ipsec_sa_t *ipsec_sa, uint32_t len,
 
 	status->all |= sa_tl->lifetime_status.all;
 
+	if (is_out_inline && (sa_tl->lifetime_status.warn.soft_exp_packets ||
+			      sa_tl->lifetime_status.warn.soft_exp_bytes)) {
+		if (!sa_tl->status_notify) {
+			_odp_ipsec_status_send(ipsec_sa->queue,
+					       ODP_IPSEC_STATUS_WARN,
+					       ipsec_sa->ipsec_sa_hdl,
+					       0, sa_tl->lifetime_status.warn);
+			sa_tl->status_notify = 1;
```

Comment:
This flag is set even if the status message could not be sent due to event allocation failure or full event queue. The return value of _odp_ipsec_status_send() should be checked.

I think it would be better to put this added section in a separate function that is called by odp_ipsec_out_inline(). Then the non-inline path and the RX path does not need to be cluttered with the "is_out_inline" checks and code which is specific to the out-inline case.

Reply:
Agreed. With v5, status event is now generated directly from the odp_ipsec_out_inline().